app/frontend/src/cljs/teet/project/project_controller.cljs

```diff
@@ -220,17 +220,44 @@
       (->NavigateToStep step-label))))
 
 (extend-protocol t/Event
-  FetchRelatedFeaturesResponse
-  (process-event [{:keys [result-path geojson-path response]} app]
-    (let [geojson (js/JSON.parse response)
+  FetchFeatureCandidatesResponse
+  (process-event [{:keys [candidate-type response]} app]
+    (let [result-path (if (= candidate-type :restrictions)
```

Comment:
these could be put to a "config map" based on candidate type and destructured from there

`(def candidate-paths {:restrictions {:result-path ...} :cadastral-units {:result-path ...}})` 

and uses `(let [{:keys [result-path ...]} (candidate-paths candidate-type)`